### PR TITLE
Update go-packages.csv

### PR DESCRIPTION
**Pull request by gracewilcox**
Updating items in the csv file marked "Need Review"

### CODE REVIEW

File: _data/releases/latest/go-packages.csv [hunk 1]
```diff
@@ -1,6 +1,6 @@
 "Package","VersionGA","VersionPreview","DisplayName","ServiceName","RepoPath","MSDocs","GHDocs","Type","New","PlannedVersions","LatestGADate","FirstGADate","Support","EOLDate","Hide","Replace","ReplaceGuide","MSDocService","ServiceId","Notes"
 "sdk/data/azappconfig","","0.5.0","App Configuration","App Configuration","data/azappconfig","","","client","true","","","","","","","","","","",""
-"sdk/storage/azdatalake","","0.1.0-beta.1","Azure Storage Data Lake","Data Lake Gen2","storage/azdatalake","","","client","true","","","","","","","","","","","Needs Review"
+"sdk/storage/azdatalake","","0.1.0-beta.1","Storage - Data Lake","Storage","storage/azdatalake","","","client","true","","","","","","","","","","",""
```
Comment (RickWinter) [inline _data/releases/latest/go-packages.csv:3]: Can you confirm the name is correct.  Its listed differently in other languages.  

"azure-storage-file-datalake","12.13.0","","Storage - Files Data Lake","Storage","storage","","","client","true","","09/12/2023","03/11/2020","active","","","","","","","Data Lake Storage Gen2"

"@azure/storage-file-datalake","12.14.0","12.15.0-beta.1","Storage - Files Data Lake","Storage","storage","","","client","true","","","03/10/2020","active","","","","","","","Data Lake Storage Gen2"

**Review by RickWinter — COMMENTED** [review]
Need to verify the name for Storage entries

File: _data/releases/latest/go-packages.csv (hunk 1 above)
Comment (ronniegeraghty) [reply _data/releases/latest/go-packages.csv:3]: Rick is right here. If this library maps to [this](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/storage/data-plane/Azure.Storage.Files.DataLake) api spec then it should be `Storage - Files Data Lake`.

Comment (gracewilcox) [reply _data/releases/latest/go-packages.csv]: fixed in [e565f59](https://github.com/Azure/azure-sdk/pull/6657/commits/e565f59a1ea16f28fa599813f758c6a0fefda56d)

**Review by ronniegeraghty — APPROVED** [review]

**Pull request merged**